emission/tests/funcTests/TestFunctionTiming.py

```diff
@@ -35,207 +17,114 @@ def setUpClass(self):
-            data = {"user_id": 123, "action": "login"}
-            logging.info(f"Data retrieved: {data}")
+        with ect.Timer() as timer_both:
+            with ect.Timer() as timer_one:
+                function_one()
+            esdsq.store_dashboard_time("function_one", timer_one)
 
-            logging.info("Step 2: Data processing")
-            time.sleep(1)  # Simulate processing time
-            processed_data = {k: str(v).upper() for k, v in data.items()}
-            logging.info(f"Processed data: {processed_data}")
+            with ect.Timer() as timer_two:
+                function_two()
+            esdsq.store_dashboard_time("function_two", timer_two)
 
-            logging.info("Step 3: Storing processed data")
-            time.sleep(0.5)  # Simulate storage delay
-            logging.info("Processed data stored successfully")
+        esdsq.store_dashboard_time("functions_one_and_two", timer_both)
 
-        with test_logger("test_code_block_instrumentation"):
-            self.run_with_timing(block_name, code_block)
+        # function_one should've taken ~1000ms; function_two should've taken ~1500ms;
+        # both functions together should've taken ~2500ms
+        self.assertAlmostEqual(timer_one.elapsed_ms, 1000, delta=100)
+        self.assertAlmostEqual(timer_two.elapsed_ms, 1500, delta=100)
+        self.assertAlmostEqual(timer_both.elapsed_ms, 2500, delta=100)
+
+        # verify that entries were stored correctly
+        self.verify_stats_entries([
+            ("stats/dashboard_time", "function_one", timer_one.elapsed_ms),
+            ("stats/dashboard_time", "function_two", timer_two.elapsed_ms),
+            ("stats/dashboard_time", "functions_one_and_two", timer_both.elapsed_ms)
+        ])
 
     def test_faulty_function_timing(self):
         """
-        Test execution and timing of a faulty function that is expected to raise an exception.
+        Test execution and timing of a faulty function that is expected to
+        raise an exception.
         """
         def faulty_function():
-            logging.info("Executing faulty_function")
+            logging.debug("Executing faulty_function")
             time.sleep(1)  # Simulate processing time before failure
             raise ValueError("Simulated error in faulty_function")
 
-        with test_logger("test_faulty_function_timing"):
-            try:
-                self.run_with_timing("faulty_function", faulty_function)
-                self.fail("ValueError was not raised by faulty_function")
-            except ValueError as e:
-                self.assertEqual(str(e), "Simulated error in faulty_function")
\ No newline at end of file
+        try:
+            with ect.Timer() as timer:
+                faulty_function()
+        except ValueError as e:
+            logging.info(f"Caught expected error: {e}")
+            esdsq.store_dashboard_error('faulty_function', timer)
+            pass
```

Comment:
At least to me, a simple `try` / `except` block in the body of the test function is more comprehensible than `with self.assertRaises(ValueError)` and extracting the error message from `context.exception`

`pass` prevents the test from failing